app/src/layouts/map/lib.ts

```diff
@@ -0,0 +1,178 @@
+	geometryFormat?: GeometryFormat;
+	geometryField?: string;
+	longitudeField?: string;
+	latitudeField?: string;
+	geometryCRS?: string;
+};
+
+export type GeometryFormat = 'geojson' | 'postgis' | 'csv' | 'wkt' | 'ewkt' | 'wkb' | 'ewkb' | 'twkb' | 'lnglat';
+export type AnyGeoJSON = FeatureCollection | Feature | GeometryCollection | Geometry;
+export type AllGeoJSON = FeatureCollection & Feature & GeometryCollection & Geometry;
+type GeometryParser = (entry: any) => wkx.Geometry | undefined;
+type GeoJSONParser = (entry: any) => Geometry | GeometryCollection | undefined;
+type Coord = [number, number];
+
+function lnglatParser(options: GeometryOptions): GeometryParser {
+	return function (entry: any) {
+		const [lng, lat] = [entry[options.longitudeField!], entry[options.latitudeField!]];
+		return lng && lat && new wkx.Point(lng, lat);
+	};
+}
+function geojsonParser(options: GeometryOptions): GeometryParser {
+	return function (entry: any) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parseGeoJSON(geom);
+	};
+}
+function twkbParser(options: GeometryOptions): GeometryParser {
+	return function (entry: any) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parseTwkb(Buffer.from(geom, 'hex'));
+	};
+}
+function wktParser(options: GeometryOptions): GeometryParser {
+	return function (entry: any) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parse(geom);
+	};
+}
+function wkbParser(options: GeometryOptions): GeometryParser {
+	return function (entry: any) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parse(Buffer.from(geom, 'hex'));
+	};
+}
+function csvParser(options: GeometryOptions): GeometryParser {
+	return function (entry: any) {
+		const geom = entry[options.geometryField!];
+		return geom && new wkx.Point(...[Number(geom[0]), Number(geom[1])]);
+	};
+}
+
+export function getGeometryParser(options: GeometryOptions): GeometryParser {
+	switch (options.geometryFormat) {
+		case 'geojson':
+			return geojsonParser(options);
+		case 'lnglat':
+			return lnglatParser(options);
+		case 'twkb':
+			return twkbParser(options);
+		case 'postgis':
+		case 'ewkb':
+		case 'wkb':
+			return wkbParser(options);
+		case 'ewkt':
+		case 'wkt':
+			return wktParser(options);
+		case 'csv':
+			return csvParser(options);
+		default:
+			throw i18n.t('unimplemented_format');
+	}
+}
+
+export function getGeometrySerializer(options: GeometryOptions): (entry: AnyGeoJSON) => any {
+	switch (options.geometryFormat) {
+		case 'geojson':
+			return (entry) => wkx.Geometry.parseGeoJSON(entry).toGeoJSON();
+		case 'wkb':
+			return (entry) => wkx.Geometry.parseGeoJSON(entry).toWkb();
+		case 'ewkb':
+			return (entry) => wkx.Geometry.parseGeoJSON(entry).toEwkb();
+		case 'twkb':
+			return (entry) => wkx.Geometry.parseGeoJSON(entry).toTwkb();
+		case 'wkt':
+			return (entry) => wkx.Geometry.parseGeoJSON(entry).toWkt();
```

Comment:
I think we can simplify this by a lot by introducing (some of) these types as "actual" Directus types, instead of an option on the layout/interface. Having a Directus-native `geojson` (and .?) types should make setup a breeze

Reply:
<img width="956" alt="CleanShot 2021-04-23 at 11 33 08@2x" src="https://user-images.githubusercontent.com/9141017/115894818-b27abc80-a427-11eb-89cc-0f0b6c8d7454.png">